tests/NetEscapades.EnumGenerators.Tests/EnumGeneratorTests.cs

```diff
@@ -583,4 +586,156 @@ public enum MyEnum
         Assert.Empty(diagnostics);
         return Verifier.Verify(output, Settings());
     }
+
+    [Fact]
+    public async Task GeneratesWarningForEnumInGenericType()
+    {
+        const string input =
+            """
+            using NetEscapades.EnumGenerators;
+
+            namespace MyTestNameSpace
+            {
+                public class GenericClass<T>
+                {
+                    [EnumExtensions]
+                    public enum MyEnum
+                    {
+                        First,
+                        Second,
+                    }
+                }
+            }
+            """;
+
+        // Test analyzer produces diagnostic
+        var analyzer = new EnumInGenericTypeAnalyzer();
+        var analyzerDiagnostics = await TestHelpers.GetAnalyzerDiagnosticsAsync(analyzer, new TestHelpers.Options(input));
```

Comment:
Don't create your own helpers etc here. Update the existing TestHelpers to include the new analyzer in the compilation instead.